File: articles/post-traumatic-growth-positive-psychology/cpp/ptg_indices.cpp
#include <cmath>
#include <iostream>

struct PTGIndicators {
    double ptg_score;
    double wellbeing_score;
    double distress_score;
    double meaning_making;
    double restored_agency;
    double narrative_integration;
    double social_support;
    double context_support;
    double deliberate_rumination;
    double intrusive_rumination;
    double ongoing_stress;
    double perceived_growth;
    double corroborated_growth;
};

double integration_index(const PTGIndicators& x) {
    return (
        x.meaning_making +
        x.restored_agency +
        x.narrative_integration +
        x.social_support +
        x.context_support
    ) / 5.0;
}

double reflection_balance(const PTGIndicators& x) {
    return x.deliberate_rumination - x.intrusive_rumination;
}

double growth_distress_balance(const PTGIndicators& x) {
    return x.ptg_score + x.wellbeing_score + integration_index(x) - x.distress_score - x.ongoing_stress;
}

double growth_alignment(const PTGIndicators& x) {
    return x.perceived_growth + x.corroborated_growth - std::abs(x.perceived_growth - x.corroborated_growth);
}

int main() {
    PTGIndicators example{6.88, 6.8, 5.6, 6.9, 6.8, 6.8, 7.1, 6.9, 6.7, 5.3, 4.9, 7.1, 6.6};

    std::cout << "Integration index: " << integration_index(example) << "\n";
    std::cout << "Reflection balance: " << reflection_balance(example) << "\n";
    std::cout << "Growth-distress balance: " << growth_distress_balance(example) << "\n";
    std::cout << "Growth alignment: " << growth_alignment(example) << "\n";

    return 0;
}

File: articles/post-traumatic-growth-positive-psychology/fortran/ptg_indices.f90
program ptg_indices
  implicit none
  real :: ptg_score, wellbeing_score, distress_score
  real :: meaning_making, restored_agency, narrative_integration
  real :: social_support, context_support
  real :: deliberate_rumination, intrusive_rumination, ongoing_stress
  real :: perceived_growth, corroborated_growth
  real :: integration, reflection_balance, growth_distress_balance, growth_alignment

  ptg_score = 6.88
  wellbeing_score = 6.8
  distress_score = 5.6
  meaning_making = 6.9
  restored_agency = 6.8
  narrative_integration = 6.8
  social_support = 7.1
  context_support = 6.9
  deliberate_rumination = 6.7
  intrusive_rumination = 5.3
  ongoing_stress = 4.9
  perceived_growth = 7.1
  corroborated_growth = 6.6

  integration = (meaning_making + restored_agency + narrative_integration + social_support + context_support) / 5.0
  reflection_balance = deliberate_rumination - intrusive_rumination
  growth_distress_balance = ptg_score + wellbeing_score + integration - distress_score - ongoing_stress
  growth_alignment = perceived_growth + corroborated_growth - abs(perceived_growth - corroborated_growth)

  print *, "Integration index:", integration
  print *, "Reflection balance:", reflection_balance
  print *, "Growth-distress balance:", growth_distress_balance
  print *, "Growth alignment:", growth_alignment
end program ptg_indices
